lib/repl.js

```diff
@@ -249,7 +250,15 @@ function REPLServer(prompt,
         }
         script = vm.createScript(code, {
           filename: file,
-          displayErrors: true
+          displayErrors: true,
+          async linker(specifier, scriptOrModule) {
```

Comment:
This will only be called for top-level dynamic imports right? If so, won't `scriptOrModule` always be the top-level `script` above?

It looks like this is exactly the same sort of signature as `importModuleDynamically` in the `vm.runInThisContext` case? If so, why not give it the same name?

Reply:
linkers are always inherited if the child import doesn't have one. as for the signature, `importModuleDynamically` above was the old name, i forgot to change it, and so it appears i'm missing a test :p

Reply:
> linkers are always inherited if the child import doesn't have one

So if I return a custom VM module instance, that does not have its own dynamic import hook, then this linker will be applied? But it will still only apply to dynamic import?

Reply:
correct. vm.Module uses the linker for static and dynamic import requests, the only difference is when the function is called.

Reply:
Do you mean to say that this same function is run for resolving the static import dependencies of dynamic import modules? But then it should take a Module object as the return value not a Namespace surely?

Also this code should be altered to return module records that aren't yet executed, as it will break circular references running an explicit `loader.import` here.

Please tell me I'm misinterpreting something here.

Reply:
> vm.Module uses the linker for static and dynamic import requests, the only difference is when the function is called.

>> Do you mean to say that this same function is run for resolving the static import dependencies of dynamic import modules? But then it should take a Module object as the return value not a Namespace surely?

As far as I can tell neither of the above statements apply to this `vm.createScript` `linker` API. Please let me know if I am wrong, but it sees that the `linker` is acting only as a dynamic import hook, returning the executed namespace object.

As such, perhaps just call it `resolveDynamicImport`?

Reply:
i can call it `resolveDynamicImport` but i would like it to accept objects or vm.Module/ModuleWrap instances as return values. as to circulars, i'm not sure what you mean. we already handle dynamic import with loader.import(), this doesn't really change that